### ISSUE TITLE
File not found exception export.edn 

### ISSUE THREAD

**Issue by kstehn** [clojure-emacs/orchard#73]
## Expected behavior

No exception gets thrown and the export.edn from clojuredocs gets used when avialable.

## Actual behavior

When something like info should be evaluated bä the nrepl i get the exception:
java.io.FileNotFoundException: ?C:\Users\KStehn\AppData\Local\orchard\clojuredocs\export.edn (Syntax for filname, directoryname or volume label is wrong)

## Steps to reproduce the problem

For this iam not entirely sure sadly. 
Just using cider-nrepl inside of calva to get info about a symbol.

## Environment & Version information

- VSCode 1.38.1
- cider-nrepl 0.22.3
- Calva v2.0.40-release-candidate-3

### Clojure version

v2.0.40-release-candidate-3

### Java version

OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.1+12)

### Operating system

Windows 7

**Comment by liquidz** [comment]
@kstehn Could you tell me about follows?

* Is there `C:\Users\KStehn\AppData\Local\orchard\clojuredocs\export.edn` actually?
* Is it sure that the thrown message contain `?` before path?
  * `java.io.FileNotFoundException: ?C:\Users`

**Comment by kstehn** [comment]
To your first question:
No there was nothing even the folder orchard wasn't there.
I created this and downloaded the export.edn myself and put it in this location but still same exception.

Second:
That i don't know just copied this from my cmd terminal.

On my Windows 10 machine i dont have such problems.

**Comment by liquidz** [comment]
@kstehn Thanks!
> Second:
> That i don't know just copied this from my cmd terminal.
> On my Windows 10 machine i dont have such problems.

It may seem something wrong with `orchard.util.os/cache-dir`.
https://github.com/clojure-emacs/orchard/blob/v0.5.1/src/orchard/util/os.clj#L72-L75
But I have no way to test it in windows 7 currently...

I'll try to find the way to test.

**Comment by dogenpunk** [comment]
I'm also experiencing this issue, but when connecting remotely to an application running nrepl on Windows Server 2016. The message I receive indicates that orchard is looking in ```C:\Windows\system32\config\systemprofile\AppData\Local\orchard\clojuredocs\export.edn```. The thrown error does include a "?" before the path. I'm running this application inside WebLogic.

@liquidz I tried running the code you linked to and received this ```"C:\\Windows\\system32\\config\\systemprofile\\AppData\\Local"```. Let me know if there's anything I can do to assist.

**Comment by liquidz** [comment]
@dogenpunk Thanks for your assist!

>  I tried running the code you linked to and received this "C:\\Windows\\system32\\config\\systemprofile\\AppData\\Local". 

It seems that `orchard.util.os/cache-dir` return correct path.
So `orchard.clojuredocs/cache-file-name` should be correct...
https://github.com/clojure-emacs/orchard/blob/v0.5.1/src/orchard/clojuredocs.clj#L16-L20

* Just to clarify: what is your `orchard.clojuredocs/cache-file-name` value?
* Could you make directories by the following code?
  * https://github.com/clojure-emacs/orchard/blob/v0.5.1/src/orchard/clojuredocs.clj#L25-L27
* Could you write cache file by the following code?
  * https://github.com/clojure-emacs/orchard/blob/v0.5.1/src/orchard/clojuredocs.clj#L28
* Could you execute `(slurp orchard.clojuredocs/cache-file-name)` directly?

BTW it works fine in my windows10 environment.

**Comment by dogenpunk** [comment]
- Evaluating the code for cache-file-name results in this `"C:\\Windows\\system32\\config\\systemprofile\\AppData\\Local\\orchard\\clojuredocs\\export.edn"`
- Evaluating the following code:
```(.. (io/file cache-file-name)
        getParentFile
        mkdirs)``` returns `false`
- I am able to `spit` and `slurp` to/from the file at `cache-file-name`, however, the code for writing the cache file requires a url and I'm not sure what URL to use. I was able to do
```(->> "foo" (spit cache-file-name))``` and got "foo" back from ```(slurp cache-file-name)```

**Comment by liquidz** [comment]
@dogenpunk Thanks a lot!
You can use `orchard.clojuredocs/default-edn-file-url` as url for `slurp`.

> I am able to spit and slurp to/from the file at cache-file-name,

Writing and reading cache file seems working well.
Hmm.. I'm not sure where is the problem.

**Comment by bbatsov** [comment]
@liquidz I'm thinking that it might be a better idea to just bundle a snapshot of the file with Orchard and have only a manual option to fetch an updated version of the data (exposed via middleware). This would solve the several problems, but most notably any initial slowdown people might experience while file is being fetched or why the cache resolution times out when there's no internet.

**Comment by liquidz** [comment]
@bbatsov

> have only a manual option to fetch an updated version of the data (exposed via middleware).

Does this mean we'll have a new op for manual updates on the middleware side?

If so, the current dicision table will be like the following simply?
https://github.com/clojure-emacs/orchard/blob/v0.5.11/src/orchard/clojuredocs.clj#L85-L91

```
fetched?        | N | Y |
----------------+---+---+
use snapshot    | x |   |
use fetched     |   | x |

(No internet connection check and validity check for fetched version)
```

**Comment by bbatsov** [comment]
> Does this mean we'll have a new op for manual updates on the middleware side

We already have the `clojuredocs-refresh-cache` op and we can potentially recycle it, or we can have something named `clojuredocs-fetch-latest` or `clojuredocs-update`. Not that big of a deal.

> If so, the current dicision table will be like the following simply?

Yeah. Simplicity is what we're aiming for. :-) ClojureDocs doesn't change that much anyways.

**Comment by liquidz** [comment]
Thanks for your confirmation!
Sounds good to me😄 
I'll try to fix it.

**Comment by bbatsov** [comment]
Fix by #96
